Add missing PlayerReferences in D2k map import

penev92 opened the pull request:
Fixes #19783 by adding the relevant player references, which allows the map import process to finish. Then the user is free to manually fix the automated player references as needed.

penev92 committed:
Add missing PlayerReferences in D2k map import

MustaphaTR reviewed (dismissed):
Does the job of fixing the crash. But our TD/RA importer is capable of importing more player data (color, faction, alliances), maybe we could add that to here too while we are in it.

Edit: Alliances are defined in the .mis file, we are only converting with .map currently, so that won't work. But other 2 could.

penev92 commented:
Added faction name and color. I've always wanted to include MIS file parsing to the map importer but that was dumped many years ago as too much (maintenance) effort for little gain in favor of external tools - https://github.com/OpenRA/OpenRA/issues/4280#issuecomment-76161747
We should now be on-par with the TD/RA map importer in terms of `PlayerReference`s.
(someone please verify that the colors are OK!)

penev92 committed:
Adjusted D2k mission player colors

PlayerReference colors in D2k missions only affect chat text and minimap colors because actors use specific palette colors.
So using the colors from the original game's minimap.

penev92 commented:
Updated player colors as provided by @MustaphaTR [on Discord](https://discord.com/channels/153649279762694144/388282819371204608/953593088570454116).
Also updated existing missions with the new colors.

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9603ff3f12c193337a793c183b4808ae12d45375...2c4555dc9e9b49e38c3acc01662658df455d5cdf)
